lib/superstore/adapters/jsonb_adapter.rb

```diff
@@ -0,0 +1,210 @@
+gem 'pg'
+require 'pg'
+
+module Superstore
+  module Adapters
+    class JsonbAdapter < AbstractAdapter
+      class QueryBuilder
+        def initialize(adapter, scope)
+          @adapter  = adapter
+          @scope    = scope
+        end
+
+        def to_query
+          [
+            "SELECT #{select_string} FROM #{@scope.klass.table_name}",
+            where_string,
+            order_string,
+            limit_string
+          ].delete_if(&:blank?) * ' '
```

Comment:
I was about to say that `Array#*` is the same as `Array#join`, but then learned that it is not.

Are you suggesting to write it as
```ruby
sql = []
add_select! sql
add_where! sql
add_order! sql
add_limit! sql
sql.join(' ')
```